docs/applications_general.py

```diff
@@ -0,0 +1,203 @@
+        Y = coor.tica(X, dim=2).get_output()
+        k_means = coor.cluster_kmeans(Y, k=num_clusters)
+        centroids = get_centroids(k_means)
+
+        markov_model = msm.estimate_markov_model(kmeans.dtrajs, 100)  # 
+        
+        previous_transition_matrix = transition_matrix
+        self.transition_matrix = markov_model.get_transition_matrix()  # figure this out
+        self._is_converged = relative_entropy(self.transition_matrix, transition_matrix) < tol
+
+    def is_converged(self):
+        return self._is_converged
+    
+    def get_transition_matrix(self):
+        return self.transition_matrix
+
+# Assuming MSMAnalyzer is an existing tool we do not want to modify,
+# create a scalems compatible operation by wrapping with a provided utility.
+
+msm_analyzer = scalems.make_operation(MSMAnalyzer,
+                                      inputs=['topfile', 'trajectory', 'transition_matrix', 'num_clusters'],
+                                      output=['is_converged', 'transition_matrix']
+                                      )
+
+# Simulation preparation and output
+# manipulation use command line tools. Simulation is executed with gmxapi.
+#
+# The idea would be that all wrappers return similar objects, so that they could be used 
+# interchangeably by the rest of the tools.
+#
+# All wrappers should have, to the extent possible, the same methods.
+
+
+# gromacs_wrapper.py:
+
+import gmxapi
+import scalems
+
+def make_input(simulation_parameters = ['md.mdp'], 
+               topology = ['md.top'], 
+               initial_conformation = ['md.gro'], 
+               binary_location = 'gmx'):
+
+
+    preprocess = scalems.commandline_operation(binary_location, 'grompp',
+                                               input_files={
+            '-f': run_parameters,
+            '-p': topology_file,
+            '-c': starting_structure},
+                                               output_files={
+            '-o': scalems.OutputFile(suffix='.tpr')
+            })
+
+    # simulation object. Structured the same for all wrappers, but you can't use one from another.
+    # strongly - a simulation toolset has some function that creates an object that encapsulates 
+    # modify that input object, and convert the simulataon to back into a simulation inpput object.
+    # the pattern holds. 
+    # key thing is that the overall program flow.  You package the iniputs and pass it to the next tools. 
+
+    # what we havee been doing with gmxapi. 
+    # 'Make input' command was consuming one file.  
+    #
+    # add features that if it is expecting 1 tpr and you give it 10 tpr files.
+    # it creates a simulatioin input object that has dynamic shape If you use it with other gmxapi components.
+    # they understand what it's shape is.   And it does support array indexing. 
+    # once you have createad aan API object. We want people to recognize it's an abstraction. 
+    # should make assumptions about accessing as an array
```

Comment:
To clarify, we should be clear that the object is not a native Python `list` or `tuple`, or even numpy array, even though we can allow subscripted indexing. This is somewhat clarified at #28 but also scattered around a bit... Unless there are objections, I'll give #28 another pass and then merge it, then try to make a sweep to refresh the various issues to clarify/coordinate discussions on object model, data model, and software components.